src/Features/Core/Portable/Common/GlyphExtensions.KnownImageIds.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+
+namespace Microsoft.CodeAnalysis;
+
+internal static partial class GlyphExtensions
+{
+    /// <summary>
+    /// This is the subset of values from <c>Microsoft.VisualStudio.Imaging.KnownImageIds</c> that we
+    /// care about. Copying them here avoids referencing Microsoft.VisualStudio.ImageCatalog.dll.
+    /// </summary>
+    private static class KnownImageIds
+    {
+        public static readonly Guid ImageCatalogGuid = Guid.Parse("ae27a6b0-e345-4288-96df-5eaf394ee369");
+
+        public const int Assembly = 196;
```

Comment:
i don't fully love this.  does this have to be defined with these same values here?  as obnoxious as it woudl be, could we not just use (or create) an enum already at this layer.  and just have a switch that converts taht to the requisite platform-specific values at that layer?

Reply:
I didn't figure you'd love this. :smile:

I think your suggestion is what Roslyn already does today. There's a switch in the EditorFeatures layer that converts the enum (`Glyph`) at the relevant host layer. The problem is that we need that information in the language server or it can't be used in Roslyn OOP. So, it can't live in the EditorFeatures layer. This is putting the logic in a central place that can be used by both the language server and the EditorFeatures layer.

These are the same values as found in Microsoft.VisualStudio.Imaging.dll.

Reply:
This _could_ definitely be moved into language server, since EditorFeatures technically depends on it, but that felt like a bit of a layering violation to me. Would you prefer that?

Reply:
tagging @dibarbet .  i think there are ways we might be able to layer this.

Reply:
That might be more appropriate that features - since it handles converting from the common Roslyn API to 'language server' types

Reply:
> `Microsoft.CodeAnalysis.LanguageServer` is not appropriate (that is the vscode exe)

Yes, sorry. That's what I meant. I was thinking of layers rather than assemblies.

> That might be more appropriate that features - since it handles converting from the common Roslyn API to 'language server' types.

In EditorFeatures, it would also be used for converting to 'VS text adornment' types. Moving to `LanguageServer.Protocol` makes sense to me. Are you OK with EditorFeatures depending on this?

Reply:
Resolution here after chatting about this in person is that I'll move this bit `LanguageServer.Protocol`.